src/components/tree/Download.tsx

```diff
@@ -0,0 +1,82 @@
+import { FC, useLayoutEffect, useMemo, useState } from "react";
+import { Link } from "react-router-dom";
+
+import useUser from "../../hooks/useUser";
+
+import { TreeResult } from "../../types/result";
+
+import DownloadIcon from "../vectors/Download";
+
+type Props = {
+ treeSections: TreeResult
+}
+
+const createColumn = (articleProperty: string | number): string => {
+ if (typeof articleProperty === "number") {
+  return String(articleProperty)
+ }
+ return articleProperty.replace(/[\[\],]/gm, "")
+}
+
+const Download: FC<Props> = ({ treeSections }) => {
+ const user = useUser();
+
+ const [uri, setUri] = useState<string>("")
+
+ const csv = useMemo(() => {
+  const articles = Object.values(treeSections).flat()
+  const headers: string = Object.keys(articles[0]).sort().join(',')
+
+  let csv: string = headers.replace(/leaf,|branch,|trunk,|root,/g, "")
+  for (const article of articles) {
+   const row: string = [
+    createColumn(article.authors.join(";") ?? ""),
+    createColumn(article.doi ?? ""),
+    createColumn(article.issue ?? ""),
+    createColumn(article.journal ?? ""),
+    createColumn(article.keywords.join(";") ?? ""),
+    createColumn(article.label ?? ""),
+    createColumn(article.page ?? ""),
+    article.title ? article.title.replaceAll(",", "") : "",
+    createColumn(article.volume ?? ""),
+    createColumn(article.year ?? ""),
+   ].join(",")
+
+   csv = csv + "\n" + row
+  }
+  return csv
+ }, [treeSections])
+
+ useLayoutEffect(() => {
+  const blob = new Blob([csv])
+  const uri = URL.createObjectURL(blob)
+  setUri(uri)
+ }, [])
```

Comment:
Aren't you missing `csv` as a dependency here?

Reply:
In this case I am passing the data through the parameters so it is not necessary to add it in the dependencies.